vitrivr-engine-module-cottontaildb/src/main/kotlin/org/vitrivr/engine/plugin/cottontaildb/descriptors/struct/StructDescriptorReader.kt

```diff
@@ -63,8 +64,13 @@ class StructDescriptorReader(field: Schema.Field<*, StructDescriptor>, connectio
 
         /* Execute query. */
         return this.connection.client.query(cottontailQuery).asSequence().map { tuple ->
-            val retrievableId = tuple.asUuidValue(RETRIEVABLE_ID_COLUMN_NAME)?.value ?: throw IllegalArgumentException("The provided tuple is missing the required field '${RETRIEVABLE_ID_COLUMN_NAME}'.")
-            val score = tuple.asDouble(SCORE_COLUMN_NAME) ?: 0.0
+            val retrievableId = tuple.asUuidValue(RETRIEVABLE_ID_COLUMN_NAME)?.value
+                ?: throw IllegalArgumentException("The provided tuple is missing the required field '${RETRIEVABLE_ID_COLUMN_NAME}'.")
+            val score = if (tuple.size > 1) {
```

Comment:
I'll pass this to @ppanopticon

Reply:
I did not yet check with @ppanopticon but read the full-text query part he implemented. The score originates there and since this is agnostic to the query, we'll have to handle this accordingly.
For now, I refactored this such that the score is only added for full-text queries.